rund.d

```diff
@@ -587,7 +587,12 @@ Flag!"compile" determineCompile(bool force, Output output, string jsonFilename,
     // TODO: what happens if json modify time is different from outputFileTime and/or build witness time?
     auto deps = readJsonFile(jsonFilename, objDir);
 
-    auto updated = deps.byKey.anyNewerThan(lastBuildTime);
+    auto updated = deps.byKey
+        // ignore the '__main.d' file. it is a temporary file
+        // generated by dmd when the -main switch is specified.
+        // it doesn't actually exist afterwards so we cannot check its timestamp
```

Comment:
Actually it doesn't ever exist on the filesystem:
https://github.com/dlang/dmd/blob/23f1f0b9ffaf7c8b5072db46168c9c519719531d/src/dmd/mars.d#L426-L433

Reply:
Got it, updated the comment